### PR TITLE
Revert "Vectorize softmax and logsoftmax (#7375)"

### PR DESCRIPTION
**Pull request by cpuhrsch**
This reverts commit 05853945a4c918dcc05139263a4f50c337b50332.

reinforce.py of [the examples](https://github.com/pytorch/examples/tree/master/reinforcement_learning) fails to converge. Working on a fix, but creating this in case that fails.

### CODE REVIEW

**Comment by cpuhrsch** [comment]
Closed since it appears close enough to naive softmax. Presumably this example needs to be tuned more. The word language model https://github.com/pytorch/examples/tree/master/word_language_model also converges and matches the CUDA version.
